tests/trainer/test_checkpoint.py

```diff
@@ -507,290 +186,333 @@ def test_checkpoint_with_object_store_logger(
-    dist.barrier()  # Ensure all ranks wrote the checkpoint file
-    save_interval_time = Time.from_timestring(save_interval)
-    if save_interval_time.unit == TimeUnit.EPOCH:
-        expected_num_checkpoints = ((num_epochs - 1) // save_interval_time.value) + 1
-    else:
-        expected_num_checkpoints = (
-            (composer_trainer_hparams.train_subset_num_batches * num_epochs - 1) // save_interval_time.value) + 1
-    checkpoint_saver = None
-    for callback in first_trainer.state.callbacks:
-        if isinstance(callback, CheckpointSaver):
-            checkpoint_saver = callback
-    assert checkpoint_saver is not None
-
-    file_was_saved = dist.get_global_rank() == 0 or deepspeed_enabled
-    if not file_was_saved:
-        expected_num_checkpoints = 0
-
-    assert len(checkpoint_saver.saved_checkpoints) == expected_num_checkpoints
-    rank_to_checkpoint_a_folder = dist.all_gather_object(os.path.abspath(checkpoint_a_folder))
-
-    checkpoint_to_resume_filepath = os.path.join(rank_to_checkpoint_a_folder[0], resume_file)
-    first_trainer_final_checkpoint_filepath = os.path.join(rank_to_checkpoint_a_folder[0], final_checkpoint)
-
-    # Move the resume and final file to the rank 0 folder
-    try:
-        rank_checkpoint_filepath = os.path.join(checkpoint_a_folder, resume_file.format(rank=dist.get_global_rank()))
-        shutil.copy2(rank_checkpoint_filepath,
-                     checkpoint_to_resume_filepath.format(rank=dist.get_global_rank()),
-                     follow_symlinks=True)
-    except (shutil.SameFileError, FileNotFoundError):
-        pass
-
-    try:
-        rank_checkpoint_filepath = os.path.join(checkpoint_a_folder,
-                                                final_checkpoint.format(rank=dist.get_global_rank()))
-        shutil.copy2(rank_checkpoint_filepath,
-                     first_trainer_final_checkpoint_filepath.format(rank=dist.get_global_rank()),
-                     follow_symlinks=True)
-    except (shutil.SameFileError, FileNotFoundError):
-        pass
-
-    checkpoint_b_folder = os.path.join(rank_to_checkpoint_a_folder[0], 'second')
-
-    second_trainer_hparams.save_folder = checkpoint_b_folder
-    second_trainer_hparams.load_path = checkpoint_to_resume_filepath
-    second_trainer_hparams.load_weights_only = False
-    second_trainer_hparams.load_strict_model_weights = False
-
-    _test_checkpoint_trainer(second_trainer_hparams)
-    dist.barrier()  # Ensure all ranks wrote the checkpoint file
-    second_trainer_final_checkpoint_filepath = os.path.join(checkpoint_b_folder, final_checkpoint)
-
-    assert_checkpoints_equivalent(
-        checkpoint_file_a=first_trainer_final_checkpoint_filepath,
-        checkpoint_file_b=second_trainer_final_checkpoint_filepath,
+        # TODO (mihir): should be checking the entire state
+        # not just the model?
+        self._assert_weights_equivalent(
+            trainer_1.state.model,
+            trainer_2.state.model,
+        )
```

Comment:
```suggestion
        self._assert_weights_equivalent(
            trainer_1.state.model,
            trainer_2.state.model,
        )
```
LGTM! yes, it should be comparing the entire state, since with `autoresume` there should be no difference from continuing the original vs. restarting from the new one